graph/pull.go

```diff
@@ -88,6 +88,10 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		logrus.Debug("image does not exist on v2 registry, falling back to v1")
 	}
 
+	if utils.DigestReference(tag) {
```

Comment:
This may not be a strong enough guarantee. Its probably better to use `digest.ParseDigest` and catch the error or `digest.DigestRegexp.Match`. Arguably, `utils.DigestReference` should not exist. The most reliable way to tell if something is a reference is to test whether it came after the `@`.

@dmcgowan